### PR TITLE
Minimum height feature for bar chart

### LINKED ISSUE #29282: [Feature]: Add barMinimumHeight property to VerticalBarChart
### Library

React / v8 (@fluentui/react)

### Describe the feature that you would like added

Currently when displaying items that have a small y-value compared to other items in a `VerticalBarChart`, the item will not display a bar or y-value which can be confused with the item having a y-value of 0. Adding `barMinimumHeight` to `IVerticalBarChartProps` would small-scale y-values to be displayed, similar to VerticalStackedBarChart.

### Have you discussed this feature with our team

_No response_

### Additional context

_No response_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Priority

Normal

### PR DESCRIPTION
**Pull request by Shubhabrata08**
## New Behavior

image

## Related Issue(s)

- Fixes #29282 

Can this be a viable way to implement this? As changing props can be a contract change for the component and might be a breaking change?
Basically setting the minimum scaled height for all non-zero values to 1.
If this is a viable way, I can proceed with updating the tests.
PS: I have quite less experience in this...

### CODE REVIEW

**Comment by Shubhabrata08** [comment]
Finalized the scaling logic
image
image
@AtishayMsft

**Comment by Shubhabrata08** [comment]
@microsoft-github-policy-service agree

**Comment by Shubhabrata08** [comment]
@AtishayMsft , I have added the changelog as well.

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 1]
```diff
@@ -536,11 +536,16 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         shouldHighlight: shouldHighlight,
       });
       const barHeight: number = Math.max(yBarScale(point.y), 0);
-      if (barHeight < 1) {
+      let adjustedBarHeight = 0;
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:539]: Thanks for fixing this Shubhabrata.

Could you also apply the same fix to the string bars below.

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 2]
```diff
@@ -536,11 +536,16 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
...
+      if (barHeight <= 0) {
         return <React.Fragment key={point.x}> </React.Fragment>;
+      } else if (barHeight <= Math.ceil(yBarScale(this._yMax) / 100.0)) {
+        adjustedBarHeight = Math.ceil(yBarScale(this._yMax) / 100.0);
+      } else {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: Are you willing to fix the same issue in vertical stacked bar chart as well.

Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: Yes... Sure... 
Will that be covered as a separate issue or will it be covered in this PR itself??

File: change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
[the remark this answers is not in the document]
Comment (Shubhabrata08) [reply change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json:2]: Happy to help... Nope... I'm not related to the team which reported this

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 1 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:539]: Sure... Just tell me if any other PR is to be created for this or should I include it in this PR itself?

Comment (AtishayMsft) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:539]: You can make all the changes in the same PR.

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 2 above)
Comment (AtishayMsft) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: You can make the changes in the same PR.

Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: Cool... Making the changes then

File: change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (AtishayMsft) [reply change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json:2]: Got it. Do let me know if you are interested to make further contributions to the fluent charting library.
We have an exciting roadmap ahead. And would be happy to recognize your contributions in the OSS community.

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 2 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: image
I have found an existing code segment which hints at a possible implementation for the feature for stacked bar charts...
Does the issue actually exist for stacked case?
Could you please have a look?

Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: Seems that it is passed as a prop??

image

File: change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Shubhabrata08) [reply change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json:2]: Sure... Would definitely love to contribute to FluentUI

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 2 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: So we wish to have a min Height for 0 value for the stacked case specially... for visual depiction of an data class existing but zero... 
Should barMinimumHeight(which is passed as prop) have the 1% height of _yMax as its lowest bound as well...

Comment (AtishayMsft) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: Yes, same behavior as vertical bar chart. Any height from 0 to 1% to have a fixed height of 1%.
Right now barHeight and barMinimumHeight are all defined in pixels not in percentages.

File: change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (AtishayMsft) [reply change/@fluentui-react-charting-8bdce3b1-3bbd-4a1d-8000-3cd8b50e1d0c.json:2]: Awesome. You can join our public discord channel https://discord.gg/9AFfTWcT and we can continue discussing there.

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 2 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:544]: Cool... So even if by passed props, barMinimumHeight is zero, we should enforce the 1% min height rule for the stacked chart?

**Comment by Shubhabrata08** [comment]
@AtishayMsft Please have a look at the changes for the Vertical stacked charts

**Comment by Shubhabrata08** [comment]
@AtishayMsft Please have a look at the changes for the Vertical stacked charts

**Comment by Shubhabrata08** [comment]
@AtishayMsft Are there any more changes required in this issue?

File: packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx [hunk 4]
```diff
@@ -810,8 +810,14 @@ export class VerticalStackedBarChartBase extends React.Component<
             />
           );
         }
-        if (barHeight < 1) {
+        let adjustedBarHeight = 0;
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx:813]: This block of change looks redundant to me. We have already modified barHeight in previous lines.

Comment (Shubhabrata08) [reply packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx]: Removed redundant section in latest push

**Review by AtishayMsft — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/401ff6d9af9536b24ab3c52c658127d19706e030'>401ff6d9af9536b24ab3c52c658127d19706e030</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Pull request merged**